src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -1021,8 +1025,34 @@ public RequestLineStatus TakeStartLine(SocketInput input)
                     scan.Skip(method.Length);
                 }
 
-                scan.Take();
+                scan.Take(); // consume space
+
+                // begin consuming request-target
                 begin = scan;
+
+                var targetBegin = scan;
+
+                if (targetBegin.Peek() != ByteForwardSlash)
+                {
+                    string requestUriScheme;
+                    if (scan.GetKnownHttpSchema(out requestUriScheme))
```

Comment:
Given what we discussed, I don't think you need this method anymore - we're not really looking for any particular schemas, since we don't do anything about them. We do want to make sure it's a valid URI, otherwise we could just accept any kind of `<garbage>/a/b/c`, and that wouldn't make sense.

So here you could check if there isn't a `/`, like you're doing, and if there isn't this is the slow path and you grab the string from `targetBegin` to the slash and check if that's a valid URI (like you did below). Otherwise you're in the hot path and you should do things as before.